### PR TITLE
[#190] Adding configuration requirements for admin usability.

### LINKED ISSUE #190: Add requirement module for easier setup after install
Follow up of #178:
`The main pain points of administrative usability have to do with all of the required configuration that needs to be done before actually configuring and using this module (set up a store, configure currencies supported in Edge, creating the products). The Kickstart install profile does a nice job of setting things up transparently, and it would be nice to move/replicate that behavior through a module setup form - see apigee/apigee-devportal-kickstart-drupal#196 (comment).`

### PR DESCRIPTION
**Pull request by arlina-espinoza**
Closes #190 .
In summary, it uses the _requirement_ module to list the "add credit" module configuration requirements as a nice report with links and configuration forms.
Note: This brings over part of the work that was done for https://github.com/apigee/apigee-devportal-kickstart-drupal/pull/241 into the `apigee_m10n_add_credit` module.
image

### CODE REVIEW

**Review by shadcn — APPROVED** [review]
This looks good to me. Thanks @arlina-espinoza

**Comment by arlina-espinoza** [comment]
Thanks @arshad. Just waiting on feedback from @cnovak regarding adding the _requirement_ module dependency on both this PR and [Kickstart #233](https://github.com/apigee/apigee-devportal-kickstart-drupal/pull/241).

**Comment by arunz6161** [comment]
Blocked by repo creation.

**Comment by arlina-espinoza** [comment]
To do: Add the drupal/requirement module to this module.

File: modules/apigee_m10n_add_credit/src/ApigeeEdgeSdkConnectorTrait.php
Comment (shadcn) [inline modules/apigee_m10n_add_credit/src/ApigeeEdgeSdkConnectorTrait.php]: @arlina-espinoza Nice 👍

**Review by shadcn — APPROVED** [review]
@arlina-espinoza This all looks good to me. Tested locally on a kickstart and non-kickstart install. LGTM (when green)

**Comment by shadcn** [comment]
@arlina-espinoza 

- I moved the Edge connection requirements to apigee_m10n since this is a requirement for apigee_m10n. 

- I also added a requirement for configuring the form display on order item. See Step 4 of https://www.drupal.org/docs/8/modules/apigee-monetization/enable-the-apigee-monetization-add-credit-module#step-3-add-products-to-the-store

Requirements_report___Apigee

File: composer.json
Comment (cnovak) [inline composer.json]: We shouldn't be adding a dev version of a module. Should also move to 1.0 as soon as possible

**Review by cnovak — CHANGES_REQUESTED** [review]
Need to use a stable version of requirement that comes from apigee/requirement-drupal project.

**Comment by cnovak** [comment]
CLAs have been signed

File: modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/AddCreditProducts.php
Comment (cnovak) [inline modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/AddCreditProducts.php]: Wrong class name

File: modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php
Comment (cnovak) [inline modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php:171]: We should follow same logic as kickstart create store form where [we use org defaults](https://github.com/apigee/apigee-devportal-kickstart-drupal/blob/768e0540ce5f2d212023a2930571393f32961491/src/Installer/Form/ApigeeMonetizationConfigurationForm.php#L311). Would be good to not replicate this form/code  in two places, any ideas?

File: modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php
Comment (cnovak) [inline modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php]: Wrong exception category

File: src/Plugin/Requirement/Requirement/ApigeeEdgeConnection.php
Comment (cnovak) [inline src/Plugin/Requirement/Requirement/ApigeeEdgeConnection.php:39]: Shouldn't this requirement check be in the Apigee Edge module instead? Or this will need to be duplicated in that module, or am I missing something?

File: src/ApigeeEdgeSdkConnectorTrait.php
Comment (cnovak) [inline src/ApigeeEdgeSdkConnectorTrait.php:8]: Shouldn't this be in Apigee Edge module? This class does not have anything monetization specific.

File: src/Plugin/Requirement/Requirement/ApigeeEdgeDeveloperSync.php
Comment (cnovak) [inline src/Plugin/Requirement/Requirement/ApigeeEdgeDeveloperSync.php:41]: Same with this class, this seems to be something that should be part of Apigee Edge module.

**Review by cnovak — CHANGES_REQUESTED** [review]

File: src/Plugin/Requirement/Requirement/ApigeeEdgeConnection.php
Comment (shadcn) [reply src/Plugin/Requirement/Requirement/ApigeeEdgeConnection.php:39]: Yes. Agreed. But we have not decided yet if we want requirements to be a dependency for `apigee_edge` so for now this lives in `apigee_m10n` but yes ideally this would be in Edge.

File: src/ApigeeEdgeSdkConnectorTrait.php
Comment (shadcn) [reply src/ApigeeEdgeSdkConnectorTrait.php:8]: See https://github.com/apigee/apigee-m10n-drupal/pull/191#discussion_r371066752

File: src/Plugin/Requirement/Requirement/ApigeeEdgeDeveloperSync.php
Comment (shadcn) [reply src/Plugin/Requirement/Requirement/ApigeeEdgeDeveloperSync.php:41]: See https://github.com/apigee/apigee-m10n-drupal/pull/191#discussion_r371066752

File: modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/AddCreditProducts.php
Comment (shadcn) [reply modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/AddCreditProducts.php]: Nice catch. Thank you 👍

File: modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php
Comment (shadcn) [reply modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php]: Fixed.

File: modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php
Comment (shadcn) [reply modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php:171]: Looking into this. Trying to figure out if there's a way we could mark requirements as an installation step.

Comment (cnovak) [reply modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php:171]: We can break this out into a separate issue, but should make sure logic is the same in both places in this PR.

File: src/Plugin/Requirement/Requirement/ApigeeEdgeConnection.php
Comment (cnovak) [reply src/Plugin/Requirement/Requirement/ApigeeEdgeConnection.php:39]: Makes sense, we can move if needed.

File: modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php
Comment (shadcn) [reply modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php:171]: OK I'll make the changes to add orgs default and we can create a separate issue for the install tasks.

Comment (shadcn) [reply modules/apigee_m10n_add_credit/src/Plugin/Requirement/Requirement/CommerceStore.php:171]: This is done and I've create an issue to handle install tasks here: https://github.com/apigee/requirement-drupal/issues/3

**Review by cnovak — APPROVED** [review]
LGTM

**Pull request merged**
